RevenueCatUI/Templates/Components/TiersComponentView.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TiersComponentView.swift
+//
+//  Created by Josh Holtz on 6/11/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+// swiftlint:disable all
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TiersComponentView: View {
+
+    let locale: Locale
+    let component: PaywallComponent.TiersComponent
+    let configuration: TemplateViewConfiguration
+
+    @State private var selectedTierIndex = 0
+
+    private var tiers: [PaywallComponent.TiersComponent.TierInfo] {
+        return component.tiers
+    }
+
+    private var componentBeforeSelector: [PaywallComponent] {
+        var selectorFound = false
+        let before = tiers[selectedTierIndex].components.filter { component in
+            if selectorFound == false {
+                if case .tierSelector(_) = component {
+                    selectorFound = true
+                } else if case .tierToggle(_) = component {
+                    selectorFound = true
+                }
+            }
+
+            return !selectorFound
+        }
+
+        return selectorFound ? before : []
+    }
+
+    private var componentAfterSelector: [PaywallComponent] {
+        var selectorFound = false
+        let after = tiers[selectedTierIndex].components.filter { component in
+            if selectorFound == false {
+                if case .tierSelector(_) = component {
+                    selectorFound = true
+                } else if case .tierToggle(_) = component {
+                    selectorFound = true
+                }
+            }
+
+            return selectorFound
+        }
+
+        return selectorFound ? after : tiers[selectedTierIndex].components
+    }
```

Comment:
similarly here, this could use `suffix(from:)`, which can potentially be more efficient if the selector is found early in  the array

```suggestion
    private var componentAfterSelector: [PaywallComponent] {
        let components = tiers[selectedTierIndex].components
        if let index = components.firstIndex(where: { 
            if case .tierSelector = $0, case .tierToggle = $0 {
                return true
            }
            return false
        }) {
            return Array(components.suffix(from: index))
        }
        return components
    }
```